src/mongocxx/test/collection.cpp

```diff
@@ -2194,27 +2194,6 @@ TEST_CASE("CRUD functionality", "[driver::collection]") {
             REQUIRE(result->inserted_count() == 1);
         }
 
-        SECTION("fail if server has maxWireVersion < 5 and write has collation") {
```

Comment:
Suggest removing other code paths that expect to run when maxWireVersion is <= 6 (for server 3.6) ([example](https://github.com/mongodb/mongo-cxx-driver/blob/dc971edbc1b37b1d7a733238966fb99c97bf3201/src/mongocxx/test/collection.cpp#L1553-L1561)):

```c++
if (test_util::get_max_wire_version(mongodb_client) >= 5) {
    // do test for MongoDB >=3.4.
} else {
    // do test for MongoDB <3.4.
}
```

Can become:
```c++
// do test for MongoDB >=3.4.
```